synapse/handlers/presence.py

```diff
@@ -1318,7 +1318,7 @@ async def get_interested_parties(
 
 async def get_interested_remotes(
     store: DataStore, states: List[UserPresenceState], state_handler: StateHandler
-) -> List[Tuple[List[str], List[UserPresenceState]]]:
+) -> List[Tuple[Iterable[str], List[UserPresenceState]]]:
```

Comment:
I applied this...and then realized that it doesn't work:

> synapse/handlers/presence.py:1352: error: Incompatible return value type (got "List[Tuple[Iterable[str], List[UserPresenceState]]]", expected "List[Tuple[Collection[str], List[UserPresenceState]]]")  [return-value]

It should though since the first element of the tuple is either a `List` or a `Set`? I'll look deeper into this.

Reply:
Ah, the type is explicitly declared on the return type too so I just needed to do it there too. See a4ff81c829581c5f6ac4a683be432aaab2b07549